applications/Preprocessor/CMakeLists.txt

```diff
@@ -1,44 +1,46 @@
-include_directories(. ${hpGEM_SOURCE_DIR}/kernel/)
 
-#the preprocessor does not perform floating poing arithmetic, so it doesn't need special Petsc/not petsc guards
-add_executable(Preprocessor.out
-               main.cpp
-               centaur.cpp
-               hpgem.cpp
-               ../../kernel/Logger.cpp
-               ../../kernel/Base/CommandLineOptions.cpp
-               ../../kernel/Base/MpiContainer.cpp)
-
-add_executable(ExtractSurface.out
-               extractSurface.cpp
-               centaur.cpp
-               hpgem.cpp
-               ../../kernel/Logger.cpp
-               ../../kernel/Base/CommandLineOptions.cpp
-               ../../kernel/Base/MpiContainer.cpp)
-
-set(CURRENT_TARGET Preprocessor.out)
-include(${CMAKE_SOURCE_DIR}/conf/cmake/CLionFix.cmake)
-set(CURRENT_TARGET ExtractSurface.out)
-include(${CMAKE_SOURCE_DIR}/conf/cmake/CLionFix.cmake)
-set(CURRENT_TARGET)
+add_library(Preprocessor-lib STATIC
+lib/centaur.cpp
+lib/hpgem.cpp
+lib/gmsh.cpp
+	)
+	# TODO: Lacking source/include separation
+	target_include_directories(Preprocessor-lib
+		PUBLIC
+			${CMAKE_CURRENT_SOURCE_DIR}/lib
+	)
+	target_link_libraries(Preprocessor-lib
+		PUBLIC
+            HPGEM::HPGEM)
+            
 
 if(hpGEM_USE_MPI)
-    target_link_libraries(Preprocessor.out MPI::MPI_CXX)
-    target_link_libraries(ExtractSurface.out MPI::MPI_CXX)
+    target_link_libraries(Preprocessor-lib PUBLIC MPI::MPI_CXX)
```

Comment:
It might be more practical if we fix this higher up, for example make it a public dependency in `kernel/Base`.

Reply:
I think you do in lines 12-14. So we can link either to the whole of hpGEM (seems a bit overkill, but easier), or just to the essential functions. But doing both seems to be a bit double.

Reply:
Ahh you are right, it is public. Good point. I think just linking to the whole of hpGEM is the way to go.